analyses/snv-callers/scripts/00-set_up.R

```diff
@@ -0,0 +1,144 @@
+# SNV Caller set up and functions
+#
+# 2019
+# C. Savonen for ALSF - CCDL
+#
+# Purpose: Set up for analyzing MAF files
+#
+# Establish base dir
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Import special functions
+source(file.path(root_dir, "analyses", "snv-callers", "util", "wrangle_functions.R"))
+source(file.path(root_dir, "analyses", "snv-callers", "util", "plot_functions.R"))
+
+#################################### Set Up ####################################
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Will need R.utils for zipping up the results file
+if (!("optparse" %in% installed.packages())) {
+  install.packages("optparse", repos = "http://cran.us.r-project.org")
+}
+# Will need R.utils for zipping up the results file
+if (!("R.utils" %in% installed.packages())) {
+  install.packages("R.utils", repos = "http://cran.us.r-project.org")
+}
+# Install package if not installed
+if (!("annotatr" %in% installed.packages())) {
+  BiocManager::install("annotatr")
+}
+# Install package if not installed
+if (!("data.table" %in% installed.packages())) {
+  install.packages("data.table", repos = "http://cran.us.r-project.org")
+}
+
+##################### Set base directories common file paths ###################
+# Declare base directory names
+scratch_dir <- file.path(root_dir, "scratch")
+data_dir <- file.path(root_dir, "data")
+snv_dir <- file.path(root_dir, "analyses", "snv-callers")
+
+# Directories that we will need
+base_results_dir <- file.path(snv_dir, "results")
+base_plots_dir <- file.path(snv_dir, "plots")
+cosmic_dir <- file.path(snv_dir, "cosmic")
+
+# Create these folders if they haven't been created yet
+if (!dir.exists(base_results_dir)) {
+  dir.create(base_results_dir)
+}
+if (!dir.exists(base_plots_dir)) {
+  dir.create(base_plots_dir)
+}
+if (!dir.exists(cosmic_dir)) {
+  dir.create(cosmic_dir)
+}
+
+# Declare reference file names
+original_metadata <- file.path(data_dir, "pbta-histologies.tsv")
+
+# These data were obtained from https://cancer.sanger.ac.uk/cosmic/download
+# These data are available if you register.
+# The full, unfiltered somatic mutations file CosmicMutantExport.tsv.gz for grch38
+# is used here.
+cosmic_file <- file.path(snv_dir, "CosmicMutantExport.tsv.gz")
```

Comment:
We might consider adding a check for this file to ensure the version is as expected. We could include an md5 hash of the file, checked with md5sum(). This might want to be separate from this script, perhaps in the readme that describes how to get the file.

Reply:
I think this should be done eventually, but because this whole analysis and its associated files is in a rather fluid state this might be a future to-do list item. What do you think?

Reply:
This item is [tracked here](https://github.com/cansav09/Notes-and-goals/issues/1)